src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.Net.Insertions.Api.Models;
+using Microsoft.Net.Insertions.Models;
+using Microsoft.Net.Insertions.Props.Models;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Packaging;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+namespace Microsoft.Net.Insertions.Api
+{
+    /// <summary>
+    /// Deduces the values of given props file variables by matching:
+    /// <list type="bullet">
+    /// <item> Paths to files extracted from nuget packages into the folders specified in default.config and .packageconfig files </item>
+    /// <item> Paths to files specified in swr files. </item>
+    /// </list>
+    /// </summary>
+    internal class PropsVariableDeducer
+    {
+        /// <summary>
+        /// Nuget feed used to download packages
+        /// </summary>
+        private readonly string _feed;
+
+        /// <summary>
+        /// Credentials for the feed.
+        /// Null, if no access token was provided.
+        /// </summary>
+        private readonly PackageSourceCredential? _credentials;
+
+        /// <summary>
+        /// Creates an instance of <see cref="PropsVariableDeducer"/>
+        /// </summary>
+        /// <param name="feed">Url to the feed that will be used to download packages</param>
+        /// <param name="accessToken">Token used to access private feed to download packages</param>
+        public PropsVariableDeducer(string feed, string? accessToken)
+        {
+            _feed = feed;
+
+            if (accessToken != null)
```

Comment:
`length == 0` means user *did* explicitly specify an access token, so he should be responsible for the validity of it; whereas `null` means user didn't use `-a:` switch when running the app and doesn't want to use props file editing feature. This isn't about validating the token, this is about determining if user wants to update props files or not.
Theoretically, empty string can be a valid access token in some feed. If not, it will fail during authorization anyway.